python/cudf/cudf/core/reshape.py

```diff
@@ -215,6 +230,16 @@ def concat(objs, axis=0, ignore_index=False, sort=None):
     # when axis is 1 (column) we can concat with Series and Dataframes
     if axis == 1:
 
+        if join == "inner" and typ is cudf.Series:
+            for i, obj in enumerate(objs):
+                obj.name = i
+            new_objs = (
+                objs[0]
+                .to_frame()
+                .join(objs[1].to_frame(), how="inner", sort="True")
+            )
+            return new_objs
+
         assert typs.issubset(allowed_typs)
```

Comment:
```suggestion
        if not typs.issubset(allowed_typs):
            raise TypeError("Can only concatenate Series and DataFrame objs when axis=1")
```

Since we are at this method, We'll have to change this assertion to be an informative error message instead.

Reply:
Sounds good! Thanks